moderation/punishments.go

```diff
@@ -445,7 +445,10 @@ func RemoveMemberMuteRole(config *Config, id int64, currentRoles []int64, mute M
 	defer gs.RUnlock()
 
 	botState := gs.MemberCopy(true, common.BotUser.ID)
```

Comment:
There is a rare case which causes the bot to not be cached in dstate. Experienced that in a huge guild with ~120k members. It will cause panic in every unmute attempt.

That seems to be issue with dstate as the /:guild/botmember internal api fails as well returning "Bot Member not found". Might be worth to use bot.GetMember & double check if the botState isn't nil to prevent issues.

Reply:
You are right! Should be fixed now! Thanks for letting me know :)